app/assets/javascripts/initializers/initializeCommentPreview.js

```diff
@@ -1,50 +1,55 @@
-function initializeCommentPreview() {
-  if (document.getElementById('preview-button')) {
-    document.getElementById('preview-button').onclick = handleCommentPreview;
-  }
-}
-
-function handleCommentPreview(e) {
-  e.preventDefault();
-  var form = e.target.parentElement.parentElement;
-  var textArea = form.querySelector('textarea');
-  if (textArea.value !== '') {
-    var previewDiv = form.querySelector('.comment-preview-div');
-    var previewButton = form.querySelector('.comment-action-preview');
-    if (previewButton.innerHTML.indexOf('PREVIEW') > -1) {
-      textArea.classList.toggle('preview-loading');
-      getAndShowPreview(previewDiv, textArea);
-      previewButton.innerHTML = 'MARKDOWN';
-    } else {
-      previewDiv.classList.toggle('preview-toggle');
-      textArea.classList.toggle('preview-toggle');
-      previewButton.innerHTML = 'PREVIEW';
-    }
+function getAndShowPreview(markdownPreviewPane, markdownEditor) {
+  function successCb(body) {
+    markdownPreviewPane.classList.toggle('preview-toggle');
+    markdownEditor.classList.toggle('preview-loading');
+    markdownEditor.classList.toggle('preview-toggle');
+    markdownPreviewPane.innerHTML = body.processed_html; // eslint-disable-line no-param-reassign
   }
-}
 
-function getAndShowPreview(previewDiv, textArea) {
-  var commentMarkdown = textArea.value;
-  var payload = JSON.stringify({
+  const payload = JSON.stringify({
     comment: {
-      body_markdown: commentMarkdown,
+      body_markdown: markdownEditor.value,
     },
   });
   getCsrfToken()
     .then(sendFetch('comment-preview', payload))
-    .then(function(response) {
+    .then(response => {
       return response.json();
     })
     .then(successCb)
-    .catch(function(err) {
-      console.log('error!');
-      console.log(err);
+    .catch(err => {
+      console.log('error!'); // eslint-disable-line
+      console.log(err); // eslint-disable-line no-console
     });
+}
 
-  function successCb(body) {
-    previewDiv.classList.toggle('preview-toggle');
-    textArea.classList.toggle('preview-loading');
-    textArea.classList.toggle('preview-toggle');
-    previewDiv.innerHTML = body.processed_html;
+function handleCommentPreview(event) {
+  event.preventDefault();
+  const { form } = event.target;
```

Comment:
No need to `.parent.parent.parent`. All form elements contained within a form always have a `form` property which is a reference to the `<form />` element that the form element is contained within.